x/profiles/legacy/v230/models_chain_links.go

```diff
@@ -0,0 +1,345 @@
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewChainConfig allows to build a new ChainConfig instance
+func NewChainConfig(name string) ChainConfig {
+	return ChainConfig{
+		Name: name,
+	}
+}
+
+// Validate checks the validity of the ChainConfig
+func (c ChainConfig) Validate() error {
+	if strings.TrimSpace(c.Name) == "" {
+		return fmt.Errorf("chain name cannot be empty or blank")
+	}
+	if c.Name != strings.ToLower(c.Name) {
+		return fmt.Errorf("chain name must be lowercase")
+	}
+	return nil
+}
+
+// --------------------------------------------------------------------------------------------------------------------
+
+// NewProof is a constructor function for Proof
+// nolint:interfacer
+func NewProof(pubKey cryptotypes.PubKey, signature string, plainText string) Proof {
+	pubKeyAny, err := codectypes.NewAnyWithValue(pubKey)
+	if err != nil {
+		panic("failed to pack public key to any type")
+	}
+	return Proof{
+		PubKey:    pubKeyAny,
+		Signature: signature,
+		PlainText: plainText,
+	}
+}
+
+// Validate checks the validity of the Proof
+func (p Proof) Validate() error {
+	if p.PubKey == nil {
+		return fmt.Errorf("public key field cannot be nil")
+	}
+
+	_, err := hex.DecodeString(p.Signature)
+	if err != nil {
+		return fmt.Errorf("invalid hex-encoded signature")
+	}
+
+	if strings.TrimSpace(p.PlainText) == "" {
+		return fmt.Errorf("plain text cannot be empty or blank")
+	}
+
+	_, err = hex.DecodeString(p.PlainText)
+	if err != nil {
+		return fmt.Errorf("invalid hex-encoded plain text")
+	}
+
+	return nil
+}
+
+// Verify verifies the signature using the given plain text and public key.
+// It returns and error if something is invalid.
+func (p Proof) Verify(unpacker codectypes.AnyUnpacker, address AddressData) error {
+	var pubkey cryptotypes.PubKey
+	err := unpacker.UnpackAny(p.PubKey, &pubkey)
+	if err != nil {
+		return fmt.Errorf("failed to unpack the public key")
+	}
+
+	value, _ := hex.DecodeString(p.PlainText)
+
+	sig, _ := hex.DecodeString(p.Signature)
+	if !pubkey.VerifySignature(value, sig) {
+		return fmt.Errorf("failed to verify the signature")
+	}
+
+	valid, err := address.VerifyPubKey(pubkey)
+	if err != nil {
+		return err
+	}
+
+	if !valid {
+		return fmt.Errorf("invalid address and public key combination provided")
+	}
+
+	return nil
+}
+
+// UnpackInterfaces implements codectypes.UnpackInterfacesMessage
+func (p *Proof) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey cryptotypes.PubKey
+	return unpacker.UnpackAny(p.PubKey, &pubKey)
+}
+
+// --------------------------------------------------------------------------------------------------------------------
+
+// AddressData is an interface representing a generic external chain address
+type AddressData interface {
+	proto.Message
+
+	// Validate checks the validity of the AddressData
+	Validate() error
+
+	// GetValue returns the address value
+	GetValue() string
```

Comment:
```suggestion
	GetAddress() string
```
As it's more explicit

Reply:
This file is the copy paste of the current version for the v210 migration since the new updates will break the proto structure.